File: app/src/main/java/com/potato/timetable/ui/login/LoginActivity.java
package com.potato.timetable.ui.login;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import com.potato.timetable.R;
import com.potato.timetable.colleges.base.CollegeFactory;
import com.potato.timetable.util.Config;
import com.potato.timetable.util.Utils;

public class LoginActivity extends AppCompatActivity implements ItemFragment.OnListFragmentInteractionListener {

    private FragmentManager fragmentManager;
    private LoginFragment loginFragment = null;
    private ItemFragment itemFragment = null;

    public static final String EXTRA_UPDATE_TIMETABLE = "update_timetable";


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        init();

        fragmentManager = getSupportFragmentManager();//初始化管理者
        String name = Config.getCollegeName();
        if (!name.isEmpty() && CollegeFactory.getCollegeNameList().contains(name)) {
            fragmentManager.beginTransaction()
                    .add(R.id.fragment_container, getLoginFragment())
                    .commit();

        } else {
            //选择学校
            fragmentManager.beginTransaction()
                    .add(R.id.fragment_container, getItemFragment())
                    .commit();
        }
    }

    public LoginFragment getLoginFragment() {
        if (loginFragment == null) {
            loginFragment = new LoginFragment();
        }
        return loginFragment;
    }

    public ItemFragment getItemFragment() {
        if (itemFragment == null) {
            itemFragment = ItemFragment.newInstance(1);
        }
        return itemFragment;
    }

    private void init() {
        setActionBar();
        ImageView bgIv = findViewById(R.id.iv_bg);
        Utils.setBackGround(this, bgIv);

        Config.readSelectCollege(this);

    }

    private void setActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(R.string.title_activity_login);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();
        if (id == android.R.id.home) {
            finish();
        } else if (id == R.id.menu_select_college) {
            fragmentManager.beginTransaction()
                    .replace(R.id.fragment_container, getItemFragment())
                    .commit();
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onListFragmentInteraction(String item) {
        Config.setCollegeName(item);
        Config.saveSelectCollege(this);
        fragmentManager.beginTransaction()
                .replace(R.id.fragment_container, getLoginFragment())
                .commit();
    }
}

File: app/src/main/java/com/potato/timetable/ui/login/LoginActivity.kt
package com.potato.timetable.ui.login

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.potato.timetable.R
import com.potato.timetable.ui.login.login.LoginFragment
import com.potato.timetable.ui.login.register.RegisterFragment

class LoginActivity : AppCompatActivity() {

    private var _viewPager: ViewPager? = null
    private val viewPager get() = _viewPager!!
    private lateinit var fragments: Array<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        fragments = arrayOf(LoginFragment.newInstance(), RegisterFragment.newInstance())
        val sectionsPagerAdapter = SectionsPagerAdapter(this, supportFragmentManager,
                fragments,
                arrayOf("登录", "注册"))

        _viewPager = findViewById(R.id.view_pager)
        viewPager.adapter = sectionsPagerAdapter
        val tabs: TabLayout = findViewById(R.id.tabs)
        tabs.setupWithViewPager(viewPager)
    }

    fun navigate(currentItem: Int, bundle: Bundle?) {
        if (currentItem < fragments.size && currentItem >= 0) {
            if (bundle != null) {
                fragments[currentItem].arguments = bundle
            }
            viewPager.currentItem = currentItem
        }

    }
}